src/test/java/org/mule/extension/BadObjectValidationTestCase.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.junit.Assert.assertThat;
+import static org.junit.rules.ExpectedException.none;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class BadObjectValidationTestCase extends AbstractSchemaValidationTestCase {
+
+  private static final String VALIDATOR_FAIL_ON_TRAILING_TOKENS = "jsonSchemaValidator.FailOnTrailingTokens";
+
+  private String json;
+
+  @Rule
+  public ExpectedException expectedException = none();
+
+  @Override
+  protected String getConfigFile() {
+    return "object-array-validation-config.xml";
+  }
+
+  @Override
+  protected void doSetUp() throws Exception {
+    json = doGetResource("inputs/bad-object.json");
+    System.setProperty(VALIDATOR_FAIL_ON_TRAILING_TOKENS, "true");
+  }
+
+  @Test
+  public void validate() throws Exception {
```

Comment:
Please, could you include the test with the system property on false to try the error case?

Reply:
yes